src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+/**
+ * Deals with the input given by the user and making sense of it.
+ */
+public class Parser {
+    /**
+     * Processes the input given by the user.
+     * Returns the corresponding Command afterwards.
+     *
+     * @param input String representation of the input given by user.
+     * @return The corresponding Command.
+     * @throws DukeException Throw DukeException.
+     */
+    public static Command parse(String input) throws DukeException {
+        input = input.strip();
+        String[] inputs = input.split(" ");
+        String action = inputs[0];
+
+        switch (action.toLowerCase()) {
+        case "bye":
+            return new Command.ExitCommand();
+        case "list":
+            return new Command.ListCommand();
+        case "done":
+            if (inputs.length == 1) {
+                throw new DukeException("Insufficient input received! " +
+                        "Please indicate the task number of the completed task.");
+            } else {
+                try {
+                    int index = Integer.parseInt(inputs[1]);
+
+                    return new Command.DoneCommand(index);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("Invalid input! Please enter the task number after 'done'.");
+                }
+            }
```

Comment:
Perhaps it would be nice to reduce nested coding. Here I removed the else block so the try-catch block is at the same indentation as the other lines in the ```done``` block. 

```suggestion
            if (inputs.length == 1) {
                throw new DukeException("Insufficient input received! " +
                        "Please indicate the task number of the completed task.");
            } 
            
            try {
                int index = Integer.parseInt(inputs[1]);
                return new Command.DoneCommand(index);
            } catch (NumberFormatException e) {
                throw new DukeException("Invalid input! Please enter the task number after 'done'.");
            }
```